rpc_storage_handler.go

```diff
@@ -999,24 +999,67 @@ func (r *RPCStorageHandler) CheckForKeyspaceChanges(orgId string) {
 	}
 }
 
+func (r *RPCStorageHandler) getKeyForCreation(keyName string) (string, string, error) {
+	//we want the key with this prefix from the master
+	newKeyName := "apikey-" + storage.HashStr(keyName)
+	//is the key in master redis
+	value, err := RPCFuncClientSingleton.CallTimeout("GetKey", newKeyName, GlobalRPCCallTimeout)
+	if err != nil {
+		emitRPCErrorEventKv(
+			rpcFuncClientSingletonCall,
+			"GetKey",
+			err,
+			map[string]string{
+				"keyName":      keyName,
+				"fixedKeyName": newKeyName,
+			},
+		)
+		if r.IsAccessError(err) {
+			if r.Login() {
+				return r.getKeyForCreation(keyName)
+			}
+		}
+		log.Debug("Error trying to get value:", err)
+		return "", "", storage.ErrKeyNotFound
+	}
+	if config.Global().SlaveOptions.EnableRPCCache {
+		// Cache key
+		RPCGlobalCache.Set(newKeyName, value, cache.DefaultExpiration)
+	}
+	//return hash key without prefix so it doesnt get double prefixed in redis
+	return value.(string), newKeyName[7:], nil
+}
+
+func (r *RPCStorageHandler) getSessionAndCreate(keyName string) {
```

Comment:
Do not think it should be a part of `RPCStorageHandler` class. Leave it as separate helper function.

Reply:
Makes sense, yep.